### PR TITLE
test: social preview

### PR DESCRIPTION
**Pull request by levi-zustiak**
# Description
- Refactors tests for `SocialPreview`

[Link](https://3.basecamp.com/3105655/buckets/36562260/todos/7145955438) to Basecamp Todo

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx [hunk 1]
```diff
@@ -1,25 +1,81 @@
...
+jest.mock('@mui/material/useMediaQuery', () => ({
+  __esModule: true,
+  default: jest.fn()
+}))
+
+const mockUseMediaQuery = useMediaQuery as jest.MockedFunction<
+  typeof useMediaQuery
+>
+
+jest.mock('@core/journeys/ui/EditorProvider', () => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx:26]: Any special reason we need to mock this out? Can we use the `EditorProvider` instead? (example below)

File: apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx [hunk 2]
```diff
@@ -1,25 +1,81 @@
...
+      dispatch: jest.fn()
+    })
+  })
+
+  it('should render Message Post and Fab for viewports larger than mobile', () => {
+    render(
       <ThemeProvider>
         <SocialPreview />
       </ThemeProvider>
     )
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx:59]:
```suggestion
  it('should render Message Post and Fab for viewports larger than mobile', () => {
    render(
      <EditorProvider initialState={state}>
        <ThemeProvider>
          <SocialPreview />
        </ThemeProvider>
      </EditorProvider>
    )
```

**Review by csiyang — CHANGES_REQUESTED** [review]
Can you also please test the `handleSelect` in the component

File: apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx (hunk 1 above)
Comment (levi-zustiak) [reply apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx:26]: Definitely would have been the better option for the two original tests haha I am going to use the mock for the click test

Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx:26]: Do you think we can change it in this PR? For code maintainability as well as don't want someone to accidentally copy this somewhere else.

Comment (levi-zustiak) [reply apps/journeys-admin/src/components/Editor/Slider/Content/Social/SocialPreview.spec.tsx:26]: Do you have an example to mock the dispatch function for me to use then? I'm not sure how to do that without this

**Review by csiyang — APPROVED** [review]

**Pull request merged**